src/commands/shared.ts

```diff
@@ -30,34 +36,51 @@ import yargs from "yargs";
  *  to be configured
  */
 const GRANT_TIMEOUT_MILLIS = 60e3;
+// The prefix of installed SSH accounts in P0 is the provider name
+export const SUPPORTED_PROVIDERS = ["aws", "gcloud"];
 
-export type BaseSshRequest = {
+export type SshRequest = AwsSshRequest | GcpSshRequest;
+
+export type BaseAwsSshRequest = {
   linuxUserName: string;
   accountId: string;
   region: string;
   id: string;
+  type: "aws";
 };
 
-export type AwsSshRoleRequest = BaseSshRequest & { role: string; type: "role" };
-export type AwsSshIdcRequest = BaseSshRequest & {
+export type AwsSshRoleRequest = BaseAwsSshRequest & {
+  role: string;
+  subType: "role";
```

Comment:
@varunalla This was called `type` before this merge commit. I renamed it to `subType` because type have values "aws", "gcloud".

I don't see this subType used anywhere except when we build the debug commands. When we log in to AWS we check the actual install config.

Reply:
@gergas3 in the backend this will become `access`, if we want to adopt that here. I think it's a bit more descriptive than merely `subType`.